opr/oprencoding/opr.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
+package oprencoding;
+
+message ProtoOPRMin {
+    string address = 1;
+    string id = 2;
+    int32 height = 3;
+    repeated string winners = 4;
+    uint64 PEG = 5;
```

Comment:
Why not `repeated uint64 assets = 5? Then have the order fixed somewhere and read/write that array accordingly? It could even takes slightly less space isn't?

Reply:
There were some talks about that, but we wanted the asset names attached to the price. Doing it like these saves space vs an array of `{"Name":"PEG", "Price":0}`, but still keeps the asset names attached to the price. 

It makes the code a bit more tedious, and with some reflect magic it would not be that hard to iterate over. I imagine in JS it could even be simpler. Maybe we should nest the fields in a sub class/struct?

Reply:
I just tested it, currently, with a random id I choose of length 16, the entry is 893 bytes. With an array of uint64, it is 844 bytes.

If we do an array of {Name, Price}, it is 1082 bytes.
Doing a map[string]uint64 is 1082 bytes.

Reply:
> Why not `repeated uint64 assets = 5? Then have the order fixed somewhere and read/write that array accordingly? It could even takes slightly less space isn't?

the only advantage i can think of is that it's a self-describing message that other languages / tools can import without explicitly also needing the list of assets. 

any change to the assets is going to require a new opr version and activation height anyway, so i think making it `repeated uint64` is fine from that perspective

i don't think we can fit the name of the asset into the message and we shouldn't

Reply:
Think I'm in agreement about using the uint64 array rather than a map. We already have forced ordering for grading so as long as this array shares that same order, I don't think there is much of a downside. Staying under 1 EC was the primary motivation for moving to protobufs anyways, so it seems a little wasted if we used the map that pushes us back into >1 EC territory. 

Also, a minor benefit is that if (when) PEG ticker is changed again, we won't need to modify the OPR code again to accommodate.

Reply:
@WhoSoup @PaulBernier Hey guys, we have like ~750 bytes with a long DID, and will probably release a new proto spec if we update. Why not just use a map? It was 1082 before we dropped 25*8 bytes for the winners, and keep the proto def `Asset = Price` until we need to bring it under 1 EC and be smart again?

Reply:
> @WhoSoup @PaulBernier Hey guys, we have like ~750 bytes with a long DID, and will probably release a new proto spec if we update. Why not just use a map? It was 1082 before we dropped 25*8 bytes for the winners, and keep the proto def `Asset = Price` until we need to bring it under 1 EC and be smart again?

"Influencers" always say "don't compromise" :smile: So I think we should go with `repeated uint64 assets = 5` now rather than just postponing and relying on our future self being smarter

Reply:
I'll go with `repeated uint64 assets = 5`